src/core/layout/qgslayoutitemelevationprofile.cpp

```diff
@@ -491,6 +496,42 @@ void QgsLayoutItemElevationProfile::setLayers( const QList<QgsMapLayer *> &layer
   invalidateCache();
 }
 
+QList<QgsAbstractProfileSource *> QgsLayoutItemElevationProfile::sources() const
+{
+  if ( mSources.isEmpty() && !mLayers.isEmpty() )
+  {
+    // Legacy: If we have layers, extract their sources and return them.
+    // We don't set mSources here, because we want the previous check to
+    // continue failing if only layers are set.
+    // TODO: Remove in QGIS 5.0.
```

Comment:
I am not sure to follow here: what would change in QGIS 5 that would allow to remove this code?